game/game.cc

```diff
@@ -238,61 +267,200 @@ int main() {
+
+                if (numberBattling == 1) {
+                  targets.emplace_back(0);
                 } else {
-                  // output list of items
-                  // pick one from list
-                  // add to pMove vector
+                  if (selectedItem.scope == SINGLE) {
+                    // select targets (if needed)
+                    if (!currPlayer.isAlive(0)) {
+                      targets.emplace_back(1);
+                    } else if (!currPlayer.isAlive(1)) {
+                      targets.emplace_back(0);
+                    } else {
+                      // Both possible targets are still alive
+                      cout << "Which HACKMON should receive the item?" << endl;
+                      cout << "0. " << currPlayer.party.at(0)->name << endl;
+                      cout << "1. " << currPlayer.party.at(1)->name << endl;
+                      size_t targetIndex;
+                      getValidValueRange<size_t>(targetIndex, 0, 1);
+                      targets.emplace_back(targetIndex);
+                    }
+                  } else {
+                    // give item to all hackmon
+                    if (currPlayer.isAlive(0)) {
+                      targets.emplace_back(0);
+                    }
+                    if (currPlayer.isAlive(1)) {
+                      targets.emplace_back(1);
+                    }
+                  }
                 }
+                // Hard-coded logic for 1v1 and 2v2 battles ends here --------------
+
+                // Move item from player's item list to itemsThisRound vector
+                itemsThisRound.emplace_back(unique_ptr<Item> {nullptr});
+                swap(itemsThisRound.back(), currPlayer.items.at(selectedItemIndex));
+                currPlayer.items.erase(currPlayer.items.begin() + selectedItemIndex);
+                actionQueue.push(itemPriority, itemsThisRound.back().get(), &currPlayer, targets);
 
               } else {
-                // output list of hackmon from numberBattling to end
-                // pick one from list
-                // call p.swapHackmon on indices
+                // Switch Hackmon
+                cout << "Which Hackmon will you swap " << currHackmon.name << " for?" << endl;
+                for (size_t index = 0; index < numberParty - currPlayer.nextAlive; ++index) {
+                  cout << index + 1 << ". " << currPlayer.party.at(index + currPlayer.nextAlive)->name << endl;
+                }
+                size_t selectedHackmonIndex;
+                getValidValueRange<size_t>(selectedHackmonIndex, 1, numberParty - currPlayer.nextAlive);
+                selectedHackmonIndex += currPlayer.nextAlive - 1; // Set to actual index into party vector
+                actionQueue.push(switchPriority, &theSwitch, &currPlayer, vector<size_t>{h, selectedHackmonIndex});
               }
             }
           }
 
+          while (!actionQueue.isEmpty()) {
```

Comment:
can we add a quick comment before the while loop explaining that this is executing the moves?